STYLEGUIDE.md

```diff
@@ -0,0 +1,367 @@
+# Design Systems Style Guide
+
+## Table of Contents
+
+- [To Contribute](#To-Contribute)
+- [Typing](#Typing)
+- [Functions](#Functions)
+- [Variable naming](#Variable-naming)
+- [Variable placement](#Variable-placement)
+- [Whitespace and line breaks](#Whitespace-and-line-breaks)
+- [JSX/React](#JSX-React)
+- [CSS-in-JS](#CSS-in-JS)
+- [File Structure](https://github.com/mongodb/leafygreen-ui/blob/main/stories/Folder-Structure.stories.mdx)
+- [References](#References)
```

Comment:
I wonder if this would make more sense as a GH Wiki? Or maybe a separate file for each section? This feels a little long & hard to read (a bit TL;DR)

Reply:
Chatted offline, leaving this as-is for now. Can re-assess based on usage!